app.R

```diff
@@ -0,0 +1,1215 @@
+                  #                                        br(),
+                  
+                  
+                  
+                  sidebarLayout(
+                    sidebarPanel(
+                      ####select datasets
+                      selectInput(inputId="datatable", label="Select Dataset:", 
+                                  choices= c('MedGPA', 'Titanic', 'Leukemia'), 
+                                  selected = 'MedGPA'),
+                      
+                      ####variable options for 'MedGPA' dataset
+                      conditionalPanel(
+                        condition = "input.datatable == 'MedGPA'",
+                        selectInput(inputId="MedYvar", label="Select Response Variable Y",
+                                    choices = c("Acceptance"),
+                                    selected = 'Acceptance'),
+                        selectInput(inputId="MedXvar", label="Select Quantitative Predictor X",
+                                    choices = c("GPA", "MCAT", "BCPM"),
+                                    selected = 'GPA')
+                      ),
+                      
+                      ####variable option for 'Titanic' dataset
+                      conditionalPanel(
+                        condition = "input.datatable == 'Titanic'",
+                        selectInput(inputId="TitanicYvar", label="Select Response Variable Y",
+                                    choices = c("Survived"),
+                                    selected = 'Survived'),
+                        selectInput(inputId="TitanicXvar", label="Select Quantitative Predictor X",
+                                    choices = c("Age"),
+                                    selected = 'Age')
+                      ),
+                      
+                      ####variable option for 'Leukemia' dataset
+                      conditionalPanel(
+                        condition = "input.datatable == 'Leukemia'",
+                        selectInput(inputId="LeukemiaYvar", label="Select Response Variable Y",
+                                    choices = c("Status"),
+                                    selected = 'Status'),
+                        selectInput(inputId="LeukemiaXvar", label="Select Quantitative Predictor X",
+                                    choices = c("Blasts", "Age", "Infil(perceptage of infiltrate)"),
+                                    selected = "Blasts")
+                      ),
+                      
+                      ###number of groups
+                      sliderInput("ngroups", "Number of Groups (Intervals):",
+                                  min = 2, max = 8,
+                                  value = 4, step = 1),
+                      
+                      br()
+                      
+                    ),
+                    mainPanel(
+                      plotOutput("empericalLogitPlot", width = "100%")%>% withSpinner(color="#ffa500")
```

Comment:
This plot wouldn't load until I added the Stat2Data package at the beginning of the code.